File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    work_queue = deque([])
    deploy = []
    for i in range(len(progresses)):
        work_time = math.ceil((100 - progresses[i]) / speeds[i])
        
        # 가장 앞의 기능이 완료되었을 경우 배포
        if work_queue:
            if work_queue[0] < work_time:
                deploy.append(len(work_queue))
                work_queue = deque([])
            
        work_queue.append(work_time)

    if work_queue:
        deploy.append(len(work_queue))
        
    return deploy

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    Queue<Integer> queue = new LinkedList<>();
    int completeDay = 0;
    int deployDay = 0;
    int count = 0;
    
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        for(int i=0; i<progresses.length; i++) {
            completeDay = (int) Math.ceil((100 - progresses[i]) / (double) speeds[i]);
            
            if(!queue.isEmpty()){
                if(queue.peek() < completeDay) {
                    answer.add(queue.size());
                    while(!queue.isEmpty()) {
                        queue.poll();
                    }
                }
            }
            queue.add(completeDay);
        }
        
        if(!queue.isEmpty()){
            answer.add(queue.size());
        }
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}